packages/babel-parser/src/tokenizer/index.js

```diff
@@ -182,17 +182,29 @@ export default class Tokenizer extends ParserErrors {
    * @memberof Tokenizer
    */
   createLookaheadState(state: State): LookaheadState {
-    return {
-      pos: state.pos,
-      value: null,
-      type: state.type,
-      start: state.start,
-      end: state.end,
-      lastTokEnd: state.end,
-      context: [this.curContext()],
-      exprAllowed: state.exprAllowed,
-      inType: state.inType,
-    };
+    if (process.env.BABEL_8_BREAKING) {
+      const { pos, type, start, end, exprAllowed, inType } = state;
```

Comment:
Use destructuring here to avoid double object access (`state.end`) and redundant object slot initialization, see https://github.com/babel/babel/pull/13341#issuecomment-848406821